message_ix_models/tools/costs/decay.py

```diff
@@ -225,10 +224,8 @@ def _get_module_cost_reduction(
             ]
 
             # append scen_red_module to scen_red_energy
-            reduction_joined = (
-                reduction_energy._append(reduction_module)
-                .reset_index(drop=True)
-                .drop(columns=["technology_type"])
+            reduction_joined = reduction_energy._append(reduction_module).reset_index(
+                drop=True
             )
         else:
             reduction_joined = reduction_energy.copy()
```

Comment:
Sorry, but what is `._append()` here? I thought pandas deprecated and removed methods like `DataFrame.append()` [in v2.0 (2023)](https://pandas.pydata.org/docs/whatsnew/v2.0.0.html#removal-of-prior-version-deprecations-changes), and I don't see this one in the documentation.

Reply:
ah interesting - it seems that function [may](https://stackoverflow.com/a/76449334) be a "private" function that isn't supported by pandas 🙈 i'll push a commit to switch to using `concat` or something else like that!

Reply:
thanks @glatterf42 for the suggestion!

the `if...else` statement is actually not on `module == "energy"` but on `if os.path.exists(ffile)`. if the file exists, then the code reads in the file as a dataframe, deletes some rows in another dataframe, and then concats the two dataframes together as the output. if the file doesn't exist, then the output is just the second dataframe copied over. i thought about condensing the code like you suggested but seems like the `if` part requires a few more steps? 😅

Reply:
Oh, sorry, I misunderstood what was going on. I guess its technically possible to still make this one line, but I don't know if it will be easier to read. Up to you if you want to try this :)

Reply:
if it's alright, i would prefer to skip editing this part for this PR and move forward with approving/merging -- thank you! :)